subcommand/server-acl-init/command.go

```diff
@@ -378,14 +378,15 @@ func (c *Command) Run(args []string) int {
 		}
 	}
 
-	// The DNS policy is attached to the anonymous token.
-	// If performing ACL replication, we assume that the primary datacenter
-	// has already created the DNS policy and attached it to the anonymous
-	// token. We don't want to modify the DNS policy in secondary datacenters
+	// The anonymous token policy needs to be configured if using Consul DNS
+	// or Consul Connect. If performing ACL replication, we assume that the
+	// primary datacenter has already created the anonymous policy and attached
+	// it to the anonymous token.
+	// We don't want to modify the anonymous policy in secondary datacenters
 	// because it is global and we can't create separate tokens for each
 	// secondary datacenter because the anonymous token is global.
-	if c.flagAllowDNS && c.flagACLReplicationTokenFile == "" {
-		err := c.configureDNSPolicies(consulClient)
+	if (c.flagAllowDNS || c.flagCreateInjectAuthMethod) && c.flagACLReplicationTokenFile == "" {
```

Comment:
We could check if `c.flagCreateACLReplicationToken` is `true` since that would signal that they plan to do multi-dc? One danger is if they generated that token out-of-band then this wouldn't be run and multi-dc connect wouldn't work.

Reply:
Lmk what you think.

Reply:
Hmm. You have to create the anon policy only in the primary datacenter. So in order for this situation to occur, someone has to use multi-dc on k8s with ACLs turned on but not setting the `-create-token-replication` flag, right? I think it's fair to assume that most people are using it through the Helm chart. 

My thinking is that creating the anonymous policy unnecessarily may raise some concern from folks, and they might potentially want to turn it off.

Reply:
👍 done!